### ISSUE TITLE
pos_ tags are not found in TAG_MAP

### ISSUE THREAD

**Issue by zafercavdar** [explosion/spaCy#5637]
## How to reproduce the behaviour

POS tags returned by Dutch, Polish, Lithuanian and Greek pre-trained models don't have matchings in TAG_MAP.py

Example code to reproduce: (same problem occurs if `nl` is replaced with `pl`, `lt`, `el`.)

```
from spacy import load
from spacy.lang.nl.examples import sentences
from spacy.lang.nl.tag_map import TAG_MAP

nl_model = load("nl_core_news_sm")
for sentence in sentences:
     doc = nl_model(sentence)
     for token in doc:
        assert token.pos_ in TAG_MAP
```
## Your Environment

* **spaCy version:** 2.3.0
* **Platform:** Linux-4.19.76-linuxkit-x86_64-with-debian-9.11
* **Python version:** 3.6.9

**Comment by adrianeboyd** [comment]
Sorry that this hasn't been documented better so far. Since you could potentially train multiple Dutch models with different corpora using different tag sets / tag maps, the saved model includes its own tag map and the tag maps may differ from the ones in the library. Before v2.3, the tag maps were in the library were in sync with the provided models, but as of v2.3 they might differ.

The tag map maps `token.tag_` to `token.pos_`, not the other way around, so the tag map keys should be the `token.tag_` values not `token.pos_` values. If you refer to the tag map saved in the model:

```
import spacy
from spacy.lang.nl.examples import sentences

nlp = spacy.load("nl_core_news_sm")
for sentence in sentences:
    doc = nlp(sentence)
    for token in doc:
        assert token.tag_ in nlp.vocab.morphology.tag_map
```

If there are tags that are missing from tag map, the `token.pos_` will end up empty (`''`). If your tagged `doc` has non-empty `pos_` tags, then the included tag map is working as intended.

If you create a blank model with `Dutch()` or `spacy.blank("nl")` it will have the tag map from the library. If you want to train with an alternate tag map, you can use the new `--tag-map-path` option for `spacy train`.

**Comment by zafercavdar** [comment]
Thanks for the explanation. Are you planning to synchronise tag maps in the library with pre-trained models in upcoming releases like v2.3.x?

**Comment by adrianeboyd** [comment]
No, probably not, but there are a bunch of minor tag map issues that made the decisions for v2.3 kind of difficult.

My original plan was to remove all the language-specific tag maps from the library, since they are specific to a particular corpus, not really language data per se. There's no real reason for a blank model to have a tag map that's specific to a particular corpus. We also don't want to try to maintain a tag map in the core library that covers all versions of all corpora for a particular language, which was happening for some languages. It makes more sense to store a tag map along with your training data rather than in the library itself.

However, it turned out that some unit tests relied on the tag maps, so I decided removing the existing ones wasn't a priority.

It would be possible to synchronize with the provided models (obviously we have them in the models), but the internal normalized format is hard to read and will be deprecated in spacy v3, and I didn't want to spend a lot of time writing a reverse converter to make them more readable.

The internal normalized format is also a bit of hassle when exporting tag maps in JSON for the train CLI, since it includes integer keys that JSON doesn't support, so you have to convert some keys back to strings. It's all kind of non-ideal. I hope a lot of this will be straightened out in spacy v3.

Do you have a use case that relies on the tag maps in the library?

I'll add more information to the v2.3 migration guide about the tag maps soon.

**Comment by zafercavdar** [comment]
Actually my final objective while accessing tag maps is to get rid of language-specific tags and get universal & human-readable POS tags. I had modified `spacy.explain` a bit for v2.1 models to reverse convert and normalize tags.

```
from .symbols import IDS
from .util import get_lang_class

reverted_IDS = {v: k for k, v in IDS.items()}


def explain(term, lang=None):
    if term in GLOSSARY:
        return GLOSSARY[term]
    else:
        if lang is not None:
            TAG_MAP = get_lang_class(lang).TAG_MAP
            explanation = TAG_MAP.get(term)
            if explanation is not None and isinstance(explanation, dict):
                if "POS" in explanation:
                    global_tag = explanation["POS"]
                elif 74 in explanation:
                    tag_id = explanation[74]
                    global_tag = reverted_IDS[tag_id]
                else:
                    return None

                return GLOSSARY.get(global_tag)
```

If v3 will get rid of corpus-specific tags and return POS tags as in the `glossary`,  it would help me a lot.

**Comment by adrianeboyd** [comment]
I'm not sure I understand exactly what you're trying to do, but it sounds like you wish that you could customize `Token.pos_` to return more readable tags rather than the showing tags from fixed UPOS tag set?

I might suggest a custom token extension that returns the glossary definition, something like this:

```
import spacy
from spacy.tokens import Token

def get_readable_pos(token):
    return spacy.explain(token.pos_)

Token.set_extension("readable_pos", getter=get_readable_pos)

nlp = spacy.load("en_core_web_sm")

doc = nlp("This is a sentence.")
assert doc[0]._.readable_pos == "determiner"
```

**Comment by zafercavdar** [comment]
I was confused about the definition of `tag_`and `pos_` and were trying to map `tag_`'s to `pos_`'s using tag maps to be able to run `spacy.explain`. Now I understand that I can directly use `pos_`!

**Comment by adrianeboyd** [comment]
Glad to hear it's working!
